xcube/webapi/context.py

```diff
@@ -491,19 +607,15 @@ def _open_ml_dataset(self, dataset_config: DatasetConfigDict) \
                                          force_geographic=True)
                 ml_dataset = BaseMultiLevelDataset(cube, ds_id=ds_id)
         else:
-            fs_type = dataset_config.get('FileSystem', 'local')
-            if self._ml_dataset_openers \
-                    and fs_type in self._ml_dataset_openers:
-                ml_dataset_opener = self._ml_dataset_openers[fs_type]
-            elif fs_type in _MULTI_LEVEL_DATASET_OPENERS:
-                ml_dataset_opener = _MULTI_LEVEL_DATASET_OPENERS[fs_type]
-            else:
+            fs_type = dataset_config.get('FileSystem')
+            if fs_type != 'memory':
                 raise ServiceConfigError(f"Invalid FileSystem {fs_type!r}"
```

Comment:
Can we come here? How does a config look like that allows coming here?

Reply:
That would be a config that has a filesystem that is neither local nor obs nor memory, as the error message says. I just checked and found that we would already get an error earlier, as it would be attempted to create a data store from an unknown file system. That error would break the service context and prevent that any dataset can be opened. I think it would be better to change it back so that this error comes up that is specific to one data set id. Do you agree?

Reply:
Yes.